components/ticket-card.tsx

```diff
@@ -0,0 +1,196 @@
+import { format } from 'date-fns'
+import { StyleSheet, View } from 'react-native'
+import QRCode from 'react-native-qrcode-svg'
+
+import { ThemedText } from '@/components/themed-text'
+import type { UserTicket } from '@/lib/api/types'
+
+interface TicketCardProps {
+  ticket: UserTicket
+  ticketPrice?: number
+  index: number
+  totalTickets: number
+}
+
+export default function TicketCard({
+  ticket,
+  ticketPrice,
+  index,
+  totalTickets,
+}: TicketCardProps) {
+  // Format date from "YYYY-MM-DD" to "d MMM, yyyy" (e.g., "24 May, 2025")
+  const formatDate = (dateString: string): string => {
+    try {
+      const date = new Date(dateString)
+      return format(date, 'd MMM, yyyy')
+    } catch {
+      return dateString
+    }
+  }
+
+  // Format time from "HH:mm" to "h:mm a" (e.g., "8:30 AM")
+  const formatTime = (timeString: string): string => {
+    try {
+      const [hours, minutes] = timeString.split(':')
+      const date = new Date()
+      date.setHours(parseInt(hours, 10), parseInt(minutes, 10), 0, 0)
+      return format(date, 'h:mm a')
+    } catch {
+      return timeString
+    }
+  }
+
+  // Generate QR code data with all ticket information
+  // Format: Human-readable text + JSON for systems
+  const formatTimeForQR = (timeString: string): string => {
+    try {
+      const [hours, minutes] = timeString.split(':')
+      const date = new Date()
+      date.setHours(parseInt(hours, 10), parseInt(minutes, 10), 0, 0)
+      return format(date, 'h:mm a')
+    } catch {
+      return timeString
+    }
+  }
+
+  const formatDateForQR = (dateString: string): string => {
+    try {
+      const date = new Date(dateString)
+      return format(date, 'MMM d, yyyy')
+    } catch {
+      return dateString
+    }
+  }
```

Comment:
Why do we need a different format fo tht QR? Can we just use the `formatDate` from above?